Makefile

```diff
@@ -25,7 +25,7 @@ NOTIFY_CREDENTIALS ?= ~/.notify-credentials
 bootstrap: generate-version-file ## Set up everything to run the app
 	pip3 install -r requirements_for_test.txt
 	createdb notification_api || true
-	. environment.sh && flask db upgrade
+	(. environment.sh && flask db upgrade) || true
```

Comment:
I don't think this works.

I think it tries to run `. environment.sh` and it returns false so therefore does not run the `flask db upgrade` statement. Therefore meaning we don't upgrade the DB which we do want to do on concourse for our test runs

If I delete my environment.sh file and run this locally I get

```
...
createdb: database creation failed: ERROR:  database "notification_api" already exists
(. environment.sh && flask db upgrade) || true
/bin/bash: environment.sh: No such file or directory
```

If I put my environment.sh file back and run locally I get

```
...
createdb notification_api || true
createdb: database creation failed: ERROR:  database "notification_api" already exists
(. environment.sh && flask db upgrade) || true
2021-06-14T19:06:15 app app INFO no-request-id "Logging configured" [in /Users/davidmcdonald/.virtualenvs/notifications-api/lib/python3.6/site-packages/notifications_utils/logging.py:37]
```

Reply:
Oh noes! This like whack-a-mole.

> (. environment.sh || true ) && flask db upgrade

This won't work unfortunately, as the scope of commands run in `(...)` is limited to the parentheses. (I only realised this from trial and error.)

OK. I've now got something that should work in all cases:

```
(. environment.sh && flask db upgrade) || flask db upgrade
```

To spell it out and confirm I'm not missing anything:

- When running locally (assuming `environment.sh` exists), the first side of the `||` will run and thus `flask db upgrade`.
- When running locally (without `environment.sh`), the second side will run, and error with `KeyError: 'NOTIFY_ENVIRONMENT'`.
- When running on CI (without `environment.sh`), the second side will run, using the keys already in the CI environment.

So...I think this is the it. This is the one 😄.

Reply:
It wasn't the one 🤣.

In all the excitement about the logic, I forgot that CI doesn't run migrations, so that third statement is false:

> When running on CI (without environment.sh), the second side will run, using the keys already in the CI environment.

I've raised a PR to fix: https://github.com/alphagov/notifications-api/pull/3271